sdk/storage/azure-storage-common/src/file_io.cpp

```diff
@@ -22,37 +22,76 @@
 #include <windows.h>
 #endif
 
+#include <algorithm>
 #include <limits>
 #include <stdexcept>
 
 namespace Azure { namespace Storage { namespace _internal {
 
 #if defined(AZ_PLATFORM_WINDOWS)
-  FileReader::FileReader(const std::string& filename)
+  std::wstring Utf8ToWide(const std::string& narrow)
   {
     int sizeNeeded = MultiByteToWideChar(
         CP_UTF8,
         MB_ERR_INVALID_CHARS,
-        filename.data(),
-        static_cast<int>(filename.length()),
+        narrow.data(),
+        static_cast<int>(narrow.length()),
         nullptr,
         0);
     if (sizeNeeded == 0)
     {
-      throw std::runtime_error("Invalid filename.");
+      throw std::runtime_error("Failed to convert utf8 to wide chars.");
     }
-    std::wstring filenameW(sizeNeeded, L'\0');
+    std::wstring wide(sizeNeeded, L'\0');
     if (MultiByteToWideChar(
             CP_UTF8,
             MB_ERR_INVALID_CHARS,
-            filename.data(),
-            static_cast<int>(filename.length()),
-            &filenameW[0],
+            narrow.data(),
+            static_cast<int>(narrow.length()),
+            &wide[0],
             sizeNeeded)
         == 0)
     {
-      throw std::runtime_error("Invalid filename.");
+      throw std::runtime_error("Failed to convert utf8 to wide chars.");
     }
+    return wide;
+  }
```

Comment:
`open()` and `fopen()` are both deprecated on Windows in favor of the security enhancement version of them (the ones with `_s`). So I cannot get rid of the conditional compiling anyway. Moreover, it's more difficult to specify share mode with C-runtime API. So I guess I'll add some comments here so that it's easier to understand.